Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,192 @@
+## Status Subresource For Config-Based Resources
+
+* **Owners:**
+  * [yp969803](https://github.com/yp969803)
+* **Status:**
+  * `In-Progress`
+* **Related Tickets:**
+  * [#3385](https://github.com/prometheus-operator/prometheus-operator/issues/3335)
+* **Other docs:**
+  * NA
```

Comment:
We can link to the status subresource proposal for workload resources as well as the Kube API conventions doc.